solidity/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM node:15.11-alpine3.13
+
+RUN apk update
+RUN apk add --no-cache python3 make g++ curl
+
+COPY . .
+RUN npm ci
+RUN chmod -R +x scripts
+
+RUN npm run typechain
```

Comment:
Is there a way to do `npm` dependency caching here?

Reply:
Related, has this been an issue as far as timing?

Reply:
It hasn't been an issue as far as timing. Also it should be cached by the docker build if nothing before it changed

Reply:
So far the slowest parts are deploying the contracts and building the orchestrator (rust build)